### PR TITLE
Fix fetching last release

### PR DESCRIPTION
**Pull request by tgallice**
Check in the whole release to get the real last one. It's should fix the `plugin-web` build.

### CODE REVIEW

File: pkg/githubutil/releases.go [hunk 1]
```diff
@@ -13,14 +13,51 @@ import (
...
 )
 
-func GetLastVersion(repo string) (*version.Version, error) {
-	splittedRepo := strings.Split(repo, "/")
-	r, _, err := buildClient().Repositories.GetLatestRelease(
-		context.Background(),
-		splittedRepo[len(splittedRepo)-2],
-		splittedRepo[len(splittedRepo)-1],
+const (
+	maxPage = 3
```
Comment (AlexisMontagne) [inline pkg/githubutil/releases.go]: why not just 1 page of 30 items?

File: pkg/githubutil/releases.go [hunk 2]
```diff
@@ -13,14 +13,51 @@ import (
...
+			grepo,
+			&github.ListOptions{Page: page, PerPage: perPage},
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		for _, r := range rls {
+			if release == nil {
```
Comment (AlexisMontagne) [inline pkg/githubutil/releases.go]: you could just write:

```
if release != nil && strings.Compare(*r.TagName, *release.TagName) < 1 {
  continue
}

release = r
```

File: pkg/githubutil/releases.go [hunk 3]
```diff
@@ -13,14 +13,51 @@ import (
...
+				release = r
+			} else if strings.Compare(*r.TagName, *release.TagName) == 1 {
+				release = r
+			}
+		}
+
+		page = res.NextPage
```
Comment (AlexisMontagne) [inline pkg/githubutil/releases.go]: how does it handle the 0 release case?

File: pkg/githubutil/releases.go [hunk 4]
```diff
@@ -13,14 +13,51 @@ import (
...
+
+	return release, nil
+}
+
+func GetLastVersion(repo string) (*version.Version, error) {
+	r, err := GetLatestRelease(context.Background(), repo)
+
 	if err != nil {
 		if e, ok := err.(*github.ErrorResponse); ok && e.Response.StatusCode == 404 {
 			return &version.Version{Version: *semver.MustParse("v0.0.0")}, nil
```
Comment (AlexisMontagne) [inline pkg/githubutil/releases.go]: with your implementation you never could get in here. you might need to ensure r == nil && err == nil

File: pkg/githubutil/releases.go (hunk 1 above)
Comment (tgallice) [reply pkg/githubutil/releases.go]: lol yes could be :)

File: pkg/githubutil/releases.go (hunk 3 above)
Comment (tgallice) [reply pkg/githubutil/releases.go]: If nextPage is 0 this stop the loop.

**Comment by tgallice** [comment]
@AlexisMontagne I have made some changes

**Review by AlexisMontagne — APPROVED** [review]

**Pull request merged**